aya/src/util.rs

```diff
@@ -130,7 +130,7 @@ impl KernelVersion {
     }
 
     fn get_kernel_version() -> Result<Self, CurrentKernelVersionError> {
-        if let Some(v) = Self::get_ubuntu_kernel_version()? {
+        if let Some(v) = Self::get_ubuntu_kernel_version().ok().flatten() {
```

Comment:
I'm wondering if we should do the same (`ok().flatten()`) in line 142 for consistency as well? Debian kernel check probably works similarly.

Reply:
I think the debian check is less fallible since it relies on the standard kernel version

Reply:
Feel free to give suggestions on how to implement such a test, and i'll be happy to add it to the PR

Reply:
There are existing tests [here](https://github.com/aya-rs/aya/blob/b23973dd9ccc0f04d3f5916116c7cf244bc3189e/aya/src/util.rs#L382-L392)

Reply:
Ah alright, so change the `get_ubuntu_kernel_version()` to take a string and read the file outside of it? Or how would you make that testable, sorry not much experience in this.

Reply:
I don't understand what this patch is trying to achieve. Presumably your custom kernel version string is causing our version parsing to panic. The tests exercise the parser directly. You should be able to paste your string there and confirm the right thing happens.

Reply:
Right, but get_ubuntu_kernel_version() is an exception to the normal parser logic for ubuntu systems, Proxmox VE has this file (which is the only indicator for aya to know if it should read it) but its empty. My patch has nothing to to with the Tests you linked since it uses another parser that lives inside the get_ubuntu_kernel_version() method

Reply:
I see - then yes, you'll need to factor the parsing logic out so that it can be tested.

Reply:
I'm having trouble understanding what tests i am supposed to add now since i don't really change anything logic wise in the method. Testing the code i changed would require an integration test, if i understand correctly

Reply:
You say the file exists but is empty, right? so the parsing logic should handle the empty string, I guess.

Reply:
Unit tests would be awesome, but I guess they would trigger a bigger refactoring of this function.

Instead of retrieving the version by calling `Self::get_<distro>_kernel_version` inside `get_kernel_version`, we would rather want to have a function which gets the kernel string as an argument and then parses and processes it. This way we could test it in a straightforward way (by passing a string to `get_kernel_version()` and asserting the output).

On the side note, I'm not a big fan of `get_kernel_version()` utility in general. After quick grepping, I don't see it being used internally in Aya at all. The whole version parsing per distro is becoming a mess. And using kernel versions for detecting features is not even reliable - mostly because of distros often backporting quite fresh stuff to completely ancient kernels. In the long term, I think we should probably deprecate `get_kernel_version()` and instead work on some more reliable feature detection, similar to `bpftool feature`, based on:

* parsing kernel config
* compilation and loading of tiny programs, which just indicate whether a program type X and helper function Y

It would be great to start this effort as an additional crate (outside the monorepo). I have an use-case for this myself, so I'm happy to start it myself, but of course help is welcome.

Reply:
Can i make the requested changes by @tamird so that we can have tests, and can we merge this for now? I really need this.

Reply:
Yes, a larger refactoring can happen later

Reply:
@vadorovsky 
>  After quick grepping, I don't see it being used internally in Aya at all. 

We use KernelVersion::current() which calls it

Reply:
> Can i make the requested changes by @tamird so that we can have tests, and can we merge this for now? I really need this.

Sorry for late reply. Sure, go ahead just with changes requested by @tamird